src/Schema.php

```diff
@@ -36,26 +43,49 @@ public function __construct($schema)
 
     public function getPropertyDefinition($class, $property)
     {
-        // TODO: throw exception
-        $definition = &$this->schema['classes'][$class][$property];
-
-        if ($definition === null) {
-            return null;
+        $classDefinition = &$this->schema['classes'][$class];
+        $propertyDefinition = &$this->schema['classes'][$class][$property];
+
+        if ($classDefinition === null) {
+            throw new Exception(
+                self::ERROR_NO_CLASS,
+                array(
+                    'class' => $class
+                ),
+                self::ERROR_NO_CLASS . " Error: class '{$class}' does not exist."
+            );
+        }
+        
+        if ($propertyDefinition === null) {
+            throw new Exception(
+                self::ERROR_NO_PROPERTY,
+                array(
+                    'class' => $class,
+                    'property' => $property
+                ),
+                self::ERROR_NO_PROPERTY . " Error: property '{$property}' " .
+                " of class '{$class}' does not exist."
+            );
         }
 
-        $type = reset($definition);
-        $path = array_slice($definition, 1);
+        $type = reset($propertyDefinition);
+        $path = array_slice($propertyDefinition, 1);
 
         return array($type, $path);
     }
 
     public function getListDefinition($list)
     {
-        // TODO: throw exception
         $definition = &$this->schema['lists'][$list];
 
         if ($definition === null) {
-            return null;
+            throw new Exception(
+                self::ERROR_NO_LIST,
+                array(
+                    'list' => $list
+                ),
+                self::ERROR_NO_LIST. " Error: list '{$list}' does not exist."
```

Comment:
We should quote strings using a function like "json_encode". That way, we can display any string--including strings with special characters.

For example:
```php
$message = "'Hey\n'";

// Broken
$errorText = "Error: $message";
// Error: 'Hey
// '

// Safe
$messageText = json_encode($message, true);
$errorText = "Error: $messageText";
// Error: "'Hey\n'"
```